File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {

    public List<List<Integer>> combinationSum(int[] nums, int target) {
        List<List<Integer>> result = new ArrayList<>();
        combinationSum(result, 0, new ArrayList<>(), nums, target);

        return result;
    }

    public void combinationSum(List<List<Integer>> result, int start, List<Integer> current, int[] nums, int target) {
        var sum = current.stream().mapToInt(Integer::intValue).sum();
        if (sum > target) return;

        if (sum == target)
            result.add(new ArrayList<>(current));

        for (int i = start; i < nums.length; i++) {
            current.add(nums[i]);

            combinationSum(result, i, current, nums, target);

            current.remove(current.size() - 1);

        }

    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        unordered_set<int> s;
        for (int i = 0; i < nums.size() - 1; i++)
        {
            if (s.find(nums[i]) != s.end())
            {
                continue;
            }
            s.insert(nums[i]);
            for (int j = i + 1; j < nums.size(); j++)
            {
                if (nums[i] + nums[j] == target)
                    return vector<int>({i, j});
            }
        }
        return result;
    }
};
